CodeGeneration/Sources/generate-swift-syntax/templates/swiftsyntax/SyntaxEnumFile.swift

```diff
@@ -57,4 +57,47 @@ let syntaxEnumFile = SourceFileSyntax(leadingTrivia: copyrightHeader) {
       }
     }
   }
+
+  for base: SyntaxNodeKind in [.decl, .stmt, .expr, .type, .pattern] {
+    let baseName = base.rawValue.withFirstCharacterUppercased;
+    try! EnumDeclSyntax(
+      """
+      /// Enum to exhaustively switch over all different \(raw: baseName) syntax nodes.
+      public enum \(raw: baseName)SyntaxEnum
+      """
+    ) {
+      for node in NON_BASE_SYNTAX_NODES where node.base == base {
+        DeclSyntax(
+          """
+          \(node.apiAttributes())\
+          case \(node.varOrCaseName)(\(node.kind.syntaxType))
+          """
+        )
+      }
+    }
+
+    try! ExtensionDeclSyntax(
+      """
+      public extension \(raw: baseName)Syntax
```

Comment:
I believe you will be able to use the `syntaxType` type property from `SyntaxNodeKind` here. 
```suggestion
      public extension \(base.syntaxType)
```

Reply:
Nice, thanks for pointing out!